corehq/apps/export/models/new.py

```diff
@@ -1066,32 +1066,35 @@ def generate_schema_from_builds(cls, domain, app_id, identifier, force_rebuild=F
         else:
             current_schema = cls()
 
-        # These are build ids from past applications and are not necessarily the most current version
-        # They determine what the deleted items are.
-        historical_app_build_ids = cls._get_app_build_ids_to_process(
-            domain,
-            app_id,
-            current_schema.last_app_versions,
-        )
 
         # These build ids represent the current state of the app and show what items are currently
         # being used. These are essential to process.
         current_app_build_ids = cls._get_current_app_ids_for_domain(domain, app_id)
 
         current_schema = cls.update_schema_with_builds(current_schema, identifier, current_app_build_ids)
-        task = update_schema_with_builds_task.delay(
-            cls,
-            current_schema,
-            identifier,
-            historical_app_build_ids,
-        )
-        try:
-            current_schema = task.get(timeout=SCHEMA_GENERATION_TIMEOUT)
-        except TimeoutError:
-            # In this case, generating the schema has taken too long. Wait until later to
-            # generate the rest of the schema.
-            # TODO: Figure out how to properly handle this case
-            pass
+
+        # Process all historical applications if we are not currently processing them
+        if not current_schema.processing_delayed_schemas:
```

Comment:
note to self: could we ever end up in a state where `processing_delayed_schemas` is `True`, but we actually do need to process the historical applications?